go/tasks/plugins/k8s/sidecar/sidecar.go

```diff
@@ -27,6 +27,23 @@ const (
 
 type sidecarResourceHandler struct{}
 
+func mergePodSpecs(flyteConfiguredPodSpec, userSpecified *v1.PodSpec) {
```

Comment:
merge seems weird and seems will get out of date everytime we change. What if we change this so that the merging is done by the flytek8s pod_helper.

so we can have a function
flytek8s.UpdatePod(basePod v1.PodSpec)?

Reply:
yeah I agree the manual merge is clunky (see PR comment, I couldn't successfully find a way to dynamically merge the flyte defaults w/ the user specified pod spec). updated to use your suggestion